libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -0,0 +1,538 @@
+
+        mWPAndroidGlueCode.setContent(postContent);
+    }
+
+
+    /*
+        Note the way we detect we're in presence of Gutenberg blocks logic is taken from
+        https://github.com/WordPress/gutenberg/blob/5a6693589285363341bebad15bd56d9371cf8ecc/lib/register.php#L331-L345
+
+        * Determine whether a content string contains blocks. This test optimizes for
+        * performance rather than strict accuracy, detecting the pattern of a block
+        * but not validating its structure. For strict accuracy, you should use the
+        * block parser on post content.
+        *
+        * @since 1.6.0
+        * @see gutenberg_parse_blocks()
+        *
+        * @param string $content Content to test.
+        * @return bool Whether the content contains blocks.
+
+        function gutenberg_content_has_blocks( $content ) {
+            return false !== strpos( $content, '<!-- wp:' );
+        }
+     */
+    private boolean contentContainsGutenbergBlocks(String postContent) {
+        return (postContent != null && postContent.contains(GUTENBERG_BLOCK_START));
+    }
+
+    /*
+    * TODO: REMOVE THIS ONCE AZTEC COMPLETELY REPLACES THE VISUAL EDITOR IN WPANDROID APP
+     */
+    private String removeVisualEditorProgressTag(String originalText) {
+        // this regex picks any <progress> tags and any opening <span> tags for image containers
+        // as produced by the Visual Editor. Note that we don't care about closing </span> tags
+        // as the AztecParser takes care of that, and it would be very difficult to accomplish with a
+        // regex (and using a proper XML crawler would be particularly overkill)
+        if (originalText != null && originalText.contains("<progress")) {
+            String regex = "<progress.*?><\\/progress>|<span id=\"img_container.*?"
+                           + " class=\"img_container\" contenteditable=\"false\">";
+            return originalText.replaceAll(regex, "");
+        } else {
+            return originalText;
+        }
+    }
+
+    /**
+     * Returns the contents of the title field from the JavaScript editor. Should be called from a background thread
+     * where possible.
+     */
+    @Override
+    public CharSequence getTitle() {
+        if (!isAdded()) {
+            return "";
+        }
+
+        // TODO: Aztec returns a ZeroWidthJoiner when empty so, strip it. Aztec needs fixing to return empty string.
+        return StringUtils.notNullStr(mTitle.getText().toString().replaceAll("&nbsp;$", "").replaceAll("\u200B", ""));
+    }
+
+    @Override
+    public boolean isActionInProgress() {
+        return false;
+    }
+
+    /**
+     * Returns the contents of the content field from the JavaScript editor. Should be called from a background thread
+     * where possible.
+     */
+    @Override
+    public CharSequence getContent(CharSequence originalContent) {
+        return mWPAndroidGlueCode.getContent(originalContent, new OnGetContentTimeout() {
```

Comment:
Can we create a property for the `OnGetContentTimeout` so a new instance is not created each time `getContent` is called? We should be able to move the `Thread.currentThread().interrupt();` to a method which will be called through this property.